flink/v1.15/flink/src/main/java/org/apache/iceberg/flink/IcebergTableSink.java

```diff
@@ -69,16 +73,20 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context context) {
     List<String> equalityColumns =
         tableSchema.getPrimaryKey().map(UniqueConstraint::getColumns).orElseGet(ImmutableList::of);
 
-    return (DataStreamSinkProvider)
```

Comment:
I checked Flink code. Flink added a default implementation to the single abstract method in `DataStreamSinkProvider`. Adding a default implementation usually is not considered as a breaking change, except for `FunctionalInterface` in this case. But this class was never annotated with `@FunctionalInterface`.  Maybe we shouldn't have used the lambda function here at the first place.

We should fix the lambda regardless if we decided to update Flink patch version for compileOnly deps.

Reply:
Flink jira for this change: https://issues.apache.org/jira/browse/FLINK-28322

Reply:
With that, I am leaning toward not chasing the Flink patch release upgrades due to unnecessary.

Reply:
I agree with not chasing Flink patch releases if they require code changes. I'd love to make sure they compile and work, but I wouldn't spend time tracking down errors between patch versions for third-party projects. If code changes are required, then we should stop update to a new patch version.

Reply:
That's a fair point. What you're saying is that the new version of Iceberg won't work with 1.15.0, 1.15.1, and 1.15.2 anymore, so we should not update. Makes sense to me.

Reply:
@Fokko I think it is still good to update code here. We should avoid using lambda function since `DataStreamSinkProvider` is not annotated with `@FunctionalInterface`.

Reply:
@stevenzwu Got it, I've removed the version bump for now.